context/app/static/js/components/detailPage/visualization/VisualizationWrapper/VisualizationWrapper.tsx

```diff
@@ -38,6 +38,19 @@ function VisualizationWrapper({
     }),
     [isPublicationPage, shouldDisplayHeader, uuid],
   );
+  // Find parent UUID for the visualization if present
+  const parentUuid: string | undefined = useMemo(() => {
+    if (Array.isArray(vitData)) {
+      const vitDataArray = vitData as object[];
+      const found = vitDataArray.find((data) => 'parentUuid' in data) as { parentUuid: string } | undefined;
+      return found?.parentUuid;
+    }
+    if ('parentUuid' in vitData) {
+      return (vitData as { parentUuid: string }).parentUuid;
+    }
+    return undefined;
+  }, [vitData]);
```

Comment:
Since `vitData` is passed to `Visualization`, could this be calculated there instead of being passed as an extra prop?

Reply:
Good call! No need to have the extra layer of prop drilling.